pandas/core/reshape/concat.py

```diff
@@ -539,7 +542,7 @@ def __init__(
         self.new_axes = self._get_new_axes()
 
     def get_result(self):
-        cons: type[DataFrame | Series]
+        cons: Callable
```

Comment:
could annotate as Callable returning DataFrame|Series

Reply:
The `Callable` typing is actually giving errors .. (with or without return type)

Reply:
Ah, but I probably have to update the actual properties' annotation to callable as well